clang/lib/CIR/Dialect/Transforms/HoistAllocas.cpp

```diff
@@ -28,6 +28,27 @@ struct HoistAllocasPass : public HoistAllocasBase<HoistAllocasPass> {
   void runOnOperation() override;
 };
 
+static bool isOpInLoop(mlir::Operation *op) {
+  return op->getParentOfType<cir::LoopOpInterface>();
+}
+
+static bool isWhileCondition(cir::AllocaOp alloca) {
```

Comment:
Should this function be named `hasStoreToAllocaInWhileCond` or something like that?

Reply:
Updated.